### PR TITLE
Kokkos resilience

### PR DESCRIPTION
**Pull request by jeffmiles63**

### CODE REVIEW

File: core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp [hunk 1]
```diff
@@ -410,6 +410,8 @@ struct CudaParallelLaunch< DriverType
 
       KOKKOS_ENSURE_CUDA_LOCK_ARRAYS_ON_DEVICE();
 
+      printf("invoke functor on device: %ld \n", sizeof(driver));
```
Comment (mhoemmen) [inline core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp]: It's not polite for Kokkos to print something on every single process, every time a kernel launches.

File: core/unit_test/TestViewCheckpoint.hpp [hunk 2]
```diff
@@ -128,50 +126,57 @@ template < typename ExecSpace, typename CpFileSpace >
...
       view_2 = Rank2ViewType("memory_view_2", dim0, dim1);
-      view_2 = Rank2ViewType("memory_view_3", dim0, dim1);
+      typename Rank2ViewType::HostMirror h_view_2 = Kokkos::create_mirror(view_2); 
 
+      printf("create filespace view\n");
       typedef CpFileSpace cp_file_space_type;
       Kokkos::View<double**,cp_file_space_type> cp_view(file_name, dim0, dim1);
 
-      for (int i = 0; i < dim0; i++) {
+      printf("invoke initializer on device: %s \n", ExecSpace::name());
```
Comment (mhoemmen) [inline core/unit_test/TestViewCheckpoint.hpp]: You can use Kokkos profiling to log kernel launches.

File: core/src/Kokkos_View.hpp [hunk 3]
```diff
@@ -2079,9 +2081,11 @@ class View : public ViewTraits< DataType , Properties ... > {
   int use_count() const
     { return m_track.use_count(); }
 
-  KOKKOS_INLINE_FUNCTION
+  inline 
   const std::string label() const
-    { return (std::string)m_track.get_label(); }
+    { 
+       return (std::string)m_track.get_label();
```
Comment (mhoemmen) [inline core/src/Kokkos_View.hpp]: The cast should be unnecessary here.  Also, we prefer C++ - style casts to C-style casts.

File: core/src/impl/Kokkos_StdFileSpace.cpp [hunk 4]
```diff
@@ -127,7 +128,7 @@ namespace Experimental {
       std::string sFullPath = s_default_path;
       size_t pos = path.find("/");
 //      printf("adding file accessor: %s, %s, %d \n", s_default_path.c_str(), path.c_str(), (int)pos );
-      if ( pos >= 0 && pos < path.length() ) {    // only use the default if there is no path info in the path...
+      if ( (int)pos >= 0 && pos < (int)path.length() ) {    // only use the default if there is no path info in the path...
```
Comment (mhoemmen) [inline core/src/impl/Kokkos_StdFileSpace.cpp]: Prefer C++ - style casts (e.g., `static_cast`) to C-style casts.

**Comment by jeffmiles63** [comment]
closing ... reopen a new one ...

**Pull request closed without merging**